### PR TITLE
Backup: Update purchase flow to use unlinked=1

### PR DESCRIPTION
**Pull request by jwebbdev**
#### Changes proposed in this Pull Request:

* This uses the updated functionality of the Connection UI to pass a `siteProductAvailabilityHandler`

#### Jetpack product discussion

Some discussion here: https://github.com/Automattic/jetpack/issues/23171

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Apply this branch to a site (like Jurassic Ninja) via Jetpack Beta
* Ensure you can connect the site and purchase a plan (and your cart has `unlink=1` in the URL)
* Deactivate the Jetpack Backup plugin and reactivate (may need to through Jetpack Beta again)
* Reconnect the Jetpack Backup plugin and ensure you aren't brought to checkout.
* Try the same steps, but with a site with a legacy plan, to ensure you aren't put into checkout while already having a valid plan, like Jetpack Personal.

### CODE REVIEW

**Comment by jwebbdev** [comment]
I'm using the rewind state here to validate the plan.  I think the get_backup_capabilities might be a better option, but that request is done by the user, and I think we're only connected as a blog at the point we're attempting to check.

File: projects/packages/backup/src/js/hooks/useConnection.js [hunk 1]
```diff
@@ -23,6 +23,11 @@ export default function useConnection() {
 	const [ price, setPrice ] = useState( 0 );
 	const [ priceAfter, setPriceAfter ] = useState( 0 );
 
+	const checkSiteHasBackupProduct = useCallback(
+		() => apiFetch( { path: '/jetpack/v4/has-backup-plan' } ).then( res => res ),
```
Comment (kangzj) [inline projects/packages/backup/src/js/hooks/useConnection.js]: is the `.then( res => res )` necessary here?

**Review by kangzj — DISMISSED** [review]
It works well and looks good to me.

Just in case it's useful. We refactored the PHP side for Search into My Jetpack eventually, because My Jetpack needs to know whether the site has a valid plan as well in it's product card list.

https://github.com/Automattic/jetpack/blob/11af0a3407cc25e849633a764e16917e933815be/projects/packages/my-jetpack/src/products/class-search.php#L261

**Comment by jwebbdev** [comment]
> It works well and looks good to me.
> 
> Just in case it's useful. We refactored the PHP side for Search into My Jetpack eventually, because My Jetpack needs to know whether the site has a valid plan as well in it's product card list.
> 
> https://github.com/Automattic/jetpack/blob/11af0a3407cc25e849633a764e16917e933815be/projects/packages/my-jetpack/src/products/class-search.php#L261

I'll take a look at how search does that, because I think it already has similar for Backup I just wasn't sure how to utilize it.  Thanks for mentioning it.

File: projects/packages/backup/src/js/hooks/useConnection.js (hunk 1 above)
Comment (jwebbdev) [reply projects/packages/backup/src/js/hooks/useConnection.js]: Cleaned this up, it wasn't.

**Comment by sdixon194** [comment]
Having some trouble testing this one. Am I testing correctly?

- Spin up JN site with Backup and the Beta
- apply this branch to Backup
- Go to connect and checkout
- In my cart, on the checkout page, my URL is `https://wordpress.com/checkout/previous-lemming.jurassic.ninja?redirect_to=admin.php%3Fpage%3Djetpack-backup&site=previous-lemming.jurassic.ninja#step2` (I don't see an `unlink=1` option)

Do I need to sandbox the API or something?

**Comment by jwebbdev** [comment]
@sdixon194 yeah I'm looking into this.  It may be just due to the redirects in calypso still for backup. @kangzj were you getting to the unlink=1 before removing the calypso logic, or only after it was gone?

Since we already have backup in the wild this may not be quite so turn-key, but still an improvement getting to where we want.

**Comment by kangzj** [comment]
>  @kangzj were you getting to the unlink=1 before removing the calypso logic, or only after it was gone?

**Comment by kangzj** [comment]
>  @kangzj were you getting to the unlink=1 before removing the calypso logic, or only after it was gone?

I added the unlink flow before removing the calypso logic. If there are plugins in the wild, is it safe to remove the logic?

**Comment by jwebbdev** [comment]
> If there are plugins in the wild, is it safe to remove the logic?

I don't think so.  So if it's interfering in any way it's kind of a tricky situation.

**Comment by jwebbdev** [comment]
@sergeymitr Wondering if you can shed some light on the expected behavior here.  I'm noticing when I click the button to trigger the connection (Get Jetpack Backup, or Log in) I am getting `unlinked=1` in the initial URL like:

https://wordpress.com/checkout/jwebbdev.jurassic.tube/jetpack_backup_t1_yearly?redirect_to=admin.php%3Fpage%3Djetpack-backup&unlinked=1&site=jwebbdev.jurassic.tube

But then it redirects to the Connection Approval screen, and from there I actually go to the cart again and don't see `unlinked=1`.  I'm not clear on whether this is intended or a problem.  If there's an issue, I'm guessing it's with the Calypso side's old redirects, but maybe the initial URL is good enough to consider the plugin side good here?

**Comment by sergeymitr** [comment]
Hi @jwebbdev.
The PR tests well, everything works as expected.
Sorry for the delay on my part.

> But then it redirects to the Connection Approval screen, and from there I actually go to the cart again and don't see unlinked=1. I'm not clear on whether this is intended or a problem. If there's an issue, I'm guessing it's with the Calypso side's old redirects, but maybe the initial URL is good enough to consider the plugin side good here?

The `unlinked=1` parameter gets passed to Calypso in case the user stays in the "site-only" mode. It gets utilized if the user isn't logged into WPCOM, only then we offer to stay in the "site-only" mode.

As far as I see, it works as it's supposed to. I noticed an issue on the Calypso side though, going to research it a bit more.
But that has nothing to do with the PR.

**Review by sergeymitr — APPROVED** [review]
👍

**Pull request merged**